client/pages/orders/[orderId].js

```diff
@@ -303,7 +309,13 @@ export async function getServerSideProps(context) {
       };
     }
 
-    return { props: { order: existOrder } };
+    const { data: order } = await client
+      .get(`/api/orders/${orderId}`)
+      .catch((err) => {
+        console.log(err.message);
+      });
+
+    return { props: { order: order } };
```

Comment:
This is where an error occurs in the case that the user is an admin but the existOrder is undefined. 
We need to fetch the correct order and return it within props.